core/src/mender-utils.c

```diff
@@ -653,3 +653,55 @@ mender_utils_compare_wildcard(const char *str, const char *wildcard_str, bool *m
 
     return MENDER_OK;
 }
+
+bool
+mender_utils_validate_deployment_data(const cJSON *deployment_data, int version) {
+
+    assert(NULL != deployment_data);
+
+    /* When we introduce new versions of deployment data in the future, the
+     * checks below should vary based on the version passed as an argument to
+     * this function. */
+
+    struct key_and_type {
+        const char *const key;
+        cJSON_bool (*type)(const cJSON *const);
+    };
+
+    static struct key_and_type field[] = {
+        { .key = "version", .type = cJSON_IsNumber },                      /* So we can modify fields later */
+        { .key = "id", .type = cJSON_IsString },                           /* Deployment identifier */
+        { .key = "artifact_name", .type = cJSON_IsString },                /* Name of artifact */
+        { .key = "payload_types", .type = cJSON_IsArray },                 /* Types of payloads embedded in artifact */
+        { .key = "provides", .type = cJSON_IsString },                     /* Artifact provides filtered on clears provides */
+        /* { .key = "name", .type = cJSON_IsString }, */                   /* TODO: Implement (see MEN-7515) */
+        /* { .key = "reboot_requested", .type = cJSON_IsString }, */       /* TODO: Implement */
+        /* { .key = "supports_rollback", .type = cJSON_IsString }, */      /* TODO: Implement */
```

Comment:
Based on Kristian consultation in Slack I think we leave this ones out. We might find a use case where this suddenly is necessary, but then we can just add them.